src/models/TaskRun.ts

```diff
@@ -84,6 +84,6 @@ export class TaskRun implements ITaskRun {
   }
 
   public get duration(): number {
-    return this.totalRunTime ?? this.estimatedRunTime ?? 0
+    return (this.totalRunTime && this.totalRunTime > 0 ? this.totalRunTime : this.estimatedRunTime) ?? 0
```

Comment:
Changed this to address the fact that the API returns a `totalRunTime` of 0 until the run has entered a terminal state so this would never fall back to the `estimatedRunTime`

Reply:
Isn't there a duration getter on the task model? There is on a flow run. @zhen0 and I fixed this same bug with flow runs.

Reply:
On the task run model? That's what this, no? Am I missing something else?

Reply:
I'm blind 🤦‍♂️